aggregator/src/blob.rs

```diff
@@ -74,6 +74,62 @@ pub struct BatchData<const N_SNARKS: usize> {
     pub chunk_data: [Vec<u8>; N_SNARKS],
 }
 
+impl<const N_SNARKS: usize> BatchData<N_SNARKS> {
+    /// For raw batch bytes with metadata, this function segments the byte stream into chunk segments.
+    /// Metadata will be removed from the result.
+    pub fn segment_with_metadata(batch_bytes_with_metadata: Vec<u8>) -> Vec<Vec<u8>> {
+        let metadata_bytes = batch_bytes_with_metadata
+            .clone()
+            .into_iter()
+            .take(182)
+            .collect::<Vec<u8>>();
+        let batch_bytes = batch_bytes_with_metadata
+            .clone()
+            .into_iter()
+            .skip(182)
+            .collect::<Vec<u8>>();
+
+        // Decoded batch bytes require segmentation based on chunk length
+        let batch_data_len = batch_bytes.len();
+        let chunk_lens = metadata_bytes[2..]
+            .chunks(4)
```

Comment:
```suggestion
            .chunks(N_ROWS_CHUNK_SIZE)
```

There isn't such a constant defined, but would be nice to do so and replace every other place in `blob.rs` wherever we use `4`